cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/util/ClouderaManagerCommandUtil.java

```diff
@@ -46,23 +47,19 @@ private ClouderaManagerCommandApiErrorParserUtil() {
         };
     }
 
-    public static List<String> getErrors(ApiCommand apiCommand, CommandsResourceApi commandsResourceApi) {
-        List<String> errors = new LinkedList<>();
+    public static List<CommandDetails> getFailedOrActiveCommands(ApiCommand apiCommand, CommandsResourceApi commandsResourceApi) {
+        List<CommandDetails> failedCommands = new LinkedList<>();
         try {
             Optional.ofNullable(apiCommand.getChildren()).map(ApiCommandList::getItems).orElse(List.of()).stream()
                     .filter(commandFailed())
                     .map(readCommand(commandsResourceApi))
-                    .forEach(cmd -> errors.addAll(getErrors(cmd, commandsResourceApi)));
-
-            if (errors.isEmpty() && StringUtils.isNotEmpty(apiCommand.getResultMessage())) {
-                String reason = String.format("Command [%s], with id [%.0f] failed: %s",
-                        apiCommand.getName(), apiCommand.getId(), apiCommand.getResultMessage());
-                errors.add(reason);
+                    .forEach(cmd -> failedCommands.addAll(getFailedOrActiveCommands(cmd, commandsResourceApi)));
```

Comment:
this looks like a recursive method call with a CM call in it, I would avoid this if possible, this could lead to enormous amount of call to CM